data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/RequestMapperTest.java

```diff
@@ -255,6 +255,57 @@ public void shouldNotRecreateProducerWhenBootstrapServerNotChanged(
       .onFailure(context::failNow);
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldNotRecreateProducerWhenAddingNewIngressWithSameBootstrapServer(
```

Comment:
Add other cases (if don't exist):
- different bootstrap servers  and the same ContentMode
- different bootstrap servers and different ContentMode
- same bootstrap server and same content mode explicitly set to `STRUCTURED`.